service/worker/workerdeployment/drainage_workflow.go

```diff
@@ -82,11 +91,48 @@ func DrainageWorkflowWithDurations(visibilityGracePeriod, refreshInterval time.D
 			if err != nil {
 				return err
 			}
-
 			if info.Status == enumspb.VERSION_DRAINAGE_STATUS_DRAINED {
 				return nil
 			}
-			_ = workflow.Sleep(ctx, refreshInterval)
+			refresh, err := getRefreshInterval(ctx, dc, ns)
+			if err != nil {
+				return err
+			}
+			_ = workflow.Sleep(ctx, refresh)
 		}
 	}
 }
+
+func getRefreshInterval(ctx workflow.Context, dc *dynamicconfig.Collection, ns string) (time.Duration, error) {
```

Comment:
Can we consolidate all these so there's one mutablesideeffect that returns a struct of all the needed values?